app/scripts/modules/ecs/src/serverGroup/configure/wizard/capacityProvider/CapacityProvider.tsx

```diff
@@ -61,6 +101,19 @@ class EcsCapacityProvider extends React.Component<IEcsCapacityProviderProps, IEc
     this.setState({ capacityProviderStrategy: capacityProviderStrategy });
   };
 
+  private updateCapacityProviderStrategy = (targetCapacityProviderType: boolean) => {
+    this.setState({useDefaultCapacityProviders : targetCapacityProviderType});
+    this.props.notifyAngular("useDefaultCapacityProviders", targetCapacityProviderType);
+
+    if (targetCapacityProviderType && this.state.defaultCapacityProviderStrategy.length > 0) {
+        this.setState({capacityProviderStrategy : this.state.defaultCapacityProviderStrategy});
+        this.props.notifyAngular('capacityProviderStrategy', this.state.defaultCapacityProviderStrategy );
+    } else {
+      this.setState({capacityProviderStrategy : []});
+      this.props.notifyAngular('capacityProviderStrategy', []);
+    }
```

Comment:
Do you think it would be more maintainable with something like this:

```
const capacityProviderStrategy = targetCapacityProviderType ? this.state.defaultCapacityProviderStrategy : [];
this.setState({ capacityProviderStrategy : capacityProviderStrategy });
this.props.notifyAngular('capacityProviderStrategy', capacityProviderStrategy});
```

Reply:
Looks small and clean. Updated this logic.